### PR TITLE
Add bug and question templates to DALI github repo

### PR DESCRIPTION
**Pull request by JanuszL**
## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
- adds bug and question templates to DALI github repo

### Affected modules and functionalities:
- .github/

### Key points relevant for the review:
- NA

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature_request_form.yml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Feature Request Form
+description: Request new or improved functionality or changes to existing functionality
+title: "[FEA]: "
```
Comment (szalpal) [inline .github/ISSUE_TEMPLATE/feature_request_form.yml:18]: Likewise

File: .github/ISSUE_TEMPLATE/feature_request_form.yml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+  - type: dropdown
+    id: criticality
+    attributes:
+      label: How would you describe the priority of this feature request
+      options:
+        - Critical (currently preventing usage)
+        - High
+        - Medium
+        - Low (would be nice)
```
Comment (szalpal) [inline .github/ISSUE_TEMPLATE/feature_request_form.yml:46]: How about expressing the priority in more descriptive terms? My point is that "high", "low" etc is rather subjective. I propose using:
```suggestion
        - Must have (e.g. DALI adoption is impossible due to lack in functionality).
        - Should have (e.g. Adoption is possible, but the performance shortcomings make the solution inferior).
        - Nice to have (e.g. Adoption is possible, the feature will enhance the use-case even more).
```

File: .github/ISSUE_TEMPLATE/submit-question.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Submit question
+description: Ask a general question about DALI
+title: "[QST]: "
```
Comment (szalpal) [inline .github/ISSUE_TEMPLATE/submit-question.yml:18]: As above, personally I'd remove this.

File: .github/ISSUE_TEMPLATE/submit-question.yml [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    validations:
+      required: true
+
+  - type: checkboxes
+    id: terms
+    attributes:
+      label: Check for duplicates
+      options:
+        - label: I have searched the [open bugs/issues](https://github.com/NVIDIA/DALI/issues) and have found no duplicates for this bug report
+          required: true
```
Comment (szalpal) [inline .github/ISSUE_TEMPLATE/submit-question.yml:41]: Github points out the lack of newline at the end of the file. I don't know if we want it or not, just wanted to note this.

File: .github/ISSUE_TEMPLATE/feature_request_form.yml [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+      label: Additional context
+      description: Add any other context, code examples, or references to existing implementations about the feature request here. If applicable, please list the modules affected.
+
+  - type: checkboxes
+    id: terms
+    attributes:
+      label: Check for duplicates
+      options:
+        - label: I have searched the [open bugs/issues](https://github.com/NVIDIA/DALI/issues) and have found no duplicates for this bug report
+          required: true
```
Comment (szalpal) [inline .github/ISSUE_TEMPLATE/feature_request_form.yml:107]: Github points out the lack of newline at the end of the file. I don't know if we want it or not, just wanted to note this.

File: .github/ISSUE_TEMPLATE/bug_report_form.yml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+      label: Other/Misc.
+      description: Please enter any other helpful information here.
+
+  - type: checkboxes
+    id: terms
+    attributes:
+      label: Check for duplicates
+      options:
+        - label: I have searched the [open bugs/issues](https://github.com/NVIDIA/DALI/issues) and have found no duplicates for this bug report
+          required: true
```
Comment (szalpal) [inline .github/ISSUE_TEMPLATE/bug_report_form.yml:71]: Github points out the lack of newline at the end of the file. I don't know if we want it or not, just wanted to note this.

**Review by szalpal — APPROVED** [review]
I've put some suggestions, but nothing that blocks the PR

File: .github/ISSUE_TEMPLATE/bug_report_form.yml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Bug Report
+description: File a bug report
+title: "[BUG]: "
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply .github/ISSUE_TEMPLATE/bug_report_form.yml]: Done

File: .github/ISSUE_TEMPLATE/feature_request_form.yml (hunk 1 above)
Comment (JanuszL) [reply .github/ISSUE_TEMPLATE/feature_request_form.yml]: Done

File: .github/ISSUE_TEMPLATE/feature_request_form.yml (hunk 2 above)
Comment (JanuszL) [reply .github/ISSUE_TEMPLATE/feature_request_form.yml]: Done

File: .github/ISSUE_TEMPLATE/submit-question.yml (hunk 3 above)
Comment (JanuszL) [reply .github/ISSUE_TEMPLATE/submit-question.yml]: Done

File: .github/ISSUE_TEMPLATE/submit-question.yml (hunk 4 above)
Comment (JanuszL) [reply .github/ISSUE_TEMPLATE/submit-question.yml]: Done

File: .github/ISSUE_TEMPLATE/bug_report_form.yml (hunk 6 above)
Comment (JanuszL) [reply .github/ISSUE_TEMPLATE/bug_report_form.yml]: Done

**Review by banasraf — APPROVED** [review]

**Comment by stiepan** [comment]
Looks nice!

**Review by stiepan — APPROVED** [review]

**Pull request merged**